include/circt/Dialect/FIRRTL/FIRRTLDialect.h

```diff
@@ -13,6 +13,7 @@
 #ifndef CIRCT_DIALECT_FIRRTL_DIALECT_H
 #define CIRCT_DIALECT_FIRRTL_DIALECT_H
 
+#include "circt/Dialect/HW/HWOps.h"
```

Comment:
Annoyingly, I need HW's unpacked array type in order to implement this pass. I wasn't able to get this to build correctly without adding the [dependency on HW dialect here](https://github.com/llvm/circt/pull/1170/files#diff-9611a4867d191e85526599f17c4f46899ee85c9ceced6f6304b41c0e3eb98f9eR184). 

I cargo-culted this include out of ESI. I can see if there's a way to untangle this.

Reply:
Waht I'm saying is "move the #include to the .cpp file"

Reply:
That's fine for the pass, but I can't figure out how to keep HW as a dependent dialect (as specified in ODS) without something like this. I need this in scope for `Passes.h` so that its included ODS-generated `Passes.h.inc` doesn't throw errors during compile. I'll look into it.

Reply:
Moved this into `PassDetails.h`.

Reply:
Is it possible to just forward declare the class that you need?  `namespace sv { class WhateverOp; }`?  This keeps the transitive #include's down

Reply:
Yeah, works. Following how MLIR dialects handle this gets this working with the forward declaration (and including the header for `HWDialect.h` in the actual pass).